src/tsolvers/lrasolver/LRAModel.h

```diff
@@ -75,7 +71,23 @@ class LRAModel
 
     void printModelState();
 
-    void clear();
+    virtual void clear();
+};
+
+class TermLRAModel : public LRAModel
```

Comment:
The current logic of the LASolvers keeps track of the decision history in both the asserted bounds needed by the simplex algorithm and the asserted PTRefs needed by the SMT solver.  The model is needed by the simplex's internal logic but this is now made independent of the PTRefs since the simplex queries internal to LIA solver will in general create new bounds that the SMT solver does not need.

The proposed solution makes the PTRef-aware LRAModel a sub-class of the general LRA model.  Another possibility would be to introduce a new wrapper class for LRAModel that would take care of PTRefs. Since much of the logic for LRAModel integrates to the handling of the PTRefs, this seems to be a simpler solution, although possibly a little unorthodox.

Reply:
As discussed, I suggest to keep this the way it is for now, and return to the question once it's more clear how the model system should look like.